Lib/idlelib/stackviewer.py

```diff
@@ -71,7 +71,7 @@ def GetText(self):
             modname = "?"
         code = frame.f_code
         filename = code.co_filename
-        funcname = code.co_name
+        funcname = code.co_qualname
```

Comment:
As IDLE maintainer, I cannot be sure about this without trying some examples.  Even then, other changes are needed in the output and I might prefer to make those first.  In any case, a would want to backport, so a separate PR would be easier.  I have recorded this change as a possibility is my notes on this modue.  I suggest omitting this for now.

Reply:
Thanks for the feedback. I have removed the changes to IDLE and kept only tracing in this PR.